orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/SkipStageHandler.kt

```diff
@@ -37,8 +37,16 @@ class SkipStageHandler(
 ) : OrcaMessageHandler<SkipStage> {
   override fun handle(message: SkipStage) {
     message.withStage { stage ->
-      if (stage.status in setOf(RUNNING, NOT_STARTED)) {
+      if (stage.status in setOf(RUNNING, NOT_STARTED) || stage.isManuallySkipped()) {
         stage.status = SKIPPED
+        if (stage.isManuallySkipped()) {
+          stage.recursiveSyntheticStages().forEach {
+            if (it.status !in setOf(SUCCEEDED, TERMINAL, FAILED_CONTINUE)) {
+              it.status = SKIPPED
+              repository.storeStage(it)
```

Comment:
I suspect you'd want to send new messages onto the queue to skip these synthetic stages, rather than just marking them skipped?

Reply:
I'm not totally seeing why - any synthetic stage's tasks will have been completed as `SKIPPED` at this point, so there's nothing else to do but mark the stages as `SKIPPED`. Would it be more natural to bubble the `SkipStage` messages up - each synthetic stage marks itself as `SKIPPED`, then pushes a `SkipStage` message onto the queue with its parent as the next stage to be skipped?

Reply:
Oh, you're right. nvm. :)